src/screens/CreateCommunity/index.js

```diff
@@ -0,0 +1,327 @@
+import * as React from 'react';
+import _ from 'lodash';
+import {
+  Keyboard,
+  SafeAreaView,
+  StatusBar,
+  StyleSheet,
+  Text,
+  TouchableWithoutFeedback,
+  View
+} from 'react-native';
+import {showMessage} from 'react-native-flash-message';
+import {useNavigation} from '@react-navigation/core';
+
+import dimen from '../../utils/dimen';
+import {Button} from '../../components/Button';
+import {COLORS} from '../../utils/theme';
+import {Context} from '../../context';
+import {Header} from '../../components';
+import {Input} from '../../components/Input';
+import {ProgressBar} from '../../components/ProgressBar';
+import {fonts, normalizeFontSize} from '../../utils/fonts';
+import {setCapitalFirstLetter} from '../../utils/Utils';
+import {NavigationConstants} from '../../utils/constants';
+import {verifyCommunityName} from '../../service/topics';
+
+const MAXIMUM_NAME_LENGTH = 64;
+const MINIMUM_NAME_LENGTH = 3;
+
+const CreateCommunity = () => {
+  const navigation = useNavigation();
+  const [users, dispatch] = React.useContext(Context).users;
+  const [name, setName] = React.useState('');
+  const [typeFetch, setTypeFetch] = React.useState('');
+
+  const verifyNameDebounce = React.useCallback(
+    _.debounce(async (text) => {
+      if (text?.length < MINIMUM_NAME_LENGTH) return setTypeFetch('min');
+      if (text?.length > MAXIMUM_NAME_LENGTH) return setTypeFetch('max');
+      const result = await verifyCommunityName(text);
+      if (!result.available) {
+        return setTypeFetch('notavailable');
+      }
+
+      return setTypeFetch('available');
+    }, 500),
+    []
+  );
+
+  const checkName = async (v) => {
+    verifyNameDebounce.cancel();
+    const value = v.replace(/[^a-zA-Z0-9-_]/g, '');
+    setTypeFetch('typing');
+    setName(value);
+    if (value.length <= MAXIMUM_NAME_LENGTH) {
+      if (value.length >= MINIMUM_NAME_LENGTH) {
+        if (!Number.isNaN(v)) {
+          setTypeFetch('fetch');
+          verifyNameDebounce(value);
+        } else {
+          setTypeFetch('nan');
+        }
+      } else {
+        setTypeFetch('typing');
+      }
+    } else {
+      setTypeFetch('max');
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `checkName` function seems to be doing a lot of things. It's checking the length of the name, replacing characters, setting state, and calling an API. This could be broken down into smaller functions for better readability and maintainability. Also, it's not clear why you're checking if the value is NaN after replacing non-alphanumeric characters. If the intention is to check if the input is numeric, this should be done before replacing the characters.

```diff
-  const checkName = async (v) => {
-    verifyNameDebounce.cancel();
-    const value = v.replace(/[^a-zA-Z0-9-_]/g, '');
-    setTypeFetch('typing');
-    setName(value);
-    if (value.length <= MAXIMUM_NAME_LENGTH) {
-      if (value.length >= MINIMUM_NAME_LENGTH) {
-        if (!Number.isNaN(v)) {
-          setTypeFetch('fetch');
-          verifyNameDebounce(value);
-        } else {
-          setTypeFetch('nan');
-        }
-      } else {
-        setTypeFetch('typing');
-      }
-    } else {
-      setTypeFetch('max');
-    }
-  };
+  const sanitizeInput = (input) => input.replace(/[^a-zA-Z0-9-_]/g, '');
+
+  const checkLength = (input) => {
+    if (input.length < MINIMUM_NAME_LENGTH) return 'min';
+    if (input.length > MAXIMUM_NAME_LENGTH) return 'max';
+    return null;
+  };
+
+  const checkNumeric = (input) => Number.isNaN(Number(input)) ? null : 'nan';
+
+  const checkName = async (input) => {
+    verifyNameDebounce.cancel();
+    const sanitizedInput = sanitizeInput(input);
+    const lengthCheck = checkLength(sanitizedInput);
+    const numericCheck = checkNumeric(sanitizedInput);
+
+    if (lengthCheck) {
+      setTypeFetch(lengthCheck);
+    } else if (numericCheck) {
+      setTypeFetch(numericCheck);
+    } else {
+      setTypeFetch('fetch');
+      verifyNameDebounce(sanitizedInput);
+    }
+
+    setName(sanitizedInput);
+  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->